core/embed/rust/src/ui/model_tr/layout.rs

```diff
@@ -108,6 +159,146 @@ extern "C" fn new_confirm_text(n_args: usize, args: *const Obj, kwargs: *mut Map
+        let title = "Recovery seed";
+
+        let share_words_len = share_words.split(',').count();
+        let share_words_len_str: String<10> = String::from(share_words_len as i16);
+
+        let beginning_text: String<50> = util::format_string([
+            "Write down these ",
+            share_words_len_str.as_str(),
+            " words:\n\n",
+        ]);
+
+        let mut middle_words: String<500> = String::new();
+        for (index, word) in share_words.split(',').enumerate() {
+            let line: String<50> = util::format_string([
+                String::<2>::from(index as i16 + 1).as_str(),
+                ". ",
+                word,
+                "\n\n\n",
+            ]);
+
+            middle_words.push_str(&line).unwrap();
+        }
+
+        let end_text: String<50> = util::format_string([
+            "I wrote down all ",
+            share_words_len_str.as_str(),
+            " words in order.",
+        ]);
+
+        let text_to_show: String<600> = util::format_string([
+            beginning_text.as_str(),
+            middle_words.as_str(),
+            end_text.as_str(),
+        ]);
+
+        let obj = LayoutObj::new(Frame::new(
+            title,
+            ButtonPage::new(
+                Paragraphs::new().add::<theme::TRDefaultText>(theme::FONT_BOLD, text_to_show),
+                theme::BG,
+            ),
+        ))?;
+        Ok(obj.into())
+    };
+    unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
+}
+
+extern "C" fn confirm_word(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+    let block = |_args: &[Obj], kwargs: &Map| {
+        let choices: StrBuffer = kwargs.get(Qstr::MP_QSTR_choices)?.try_into()?;
+        let checked_index: u8 = kwargs.get(Qstr::MP_QSTR_checked_index)?.try_into()?;
+        let count: u8 = kwargs.get(Qstr::MP_QSTR_count)?.try_into()?;
+
+        let count_str: String<50> = String::from(count);
+        let checked_index_str: String<50> = String::from(checked_index + 1);
+
+        let prompt: String<50> = util::format_string([
+            "Select word ",
+            checked_index_str.as_str(),
+            " of ",
+            count_str.as_str(),
+            ":",
+        ]);
+
+        let words: Vec<String<50>, 24> = choices.split(',').map(String::from).collect();
+
+        let obj = LayoutObj::new(
+            ChoicePage::new(prompt, String::<50_usize>::from(""), words).into_child(),
+        )?;
+        Ok(obj.into())
+    };
+    unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
+}
+
+extern "C" fn request_word_count(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+    let block = |_args: &[Obj], kwargs: &Map| {
+        let title: StrBuffer = kwargs.get(Qstr::MP_QSTR_title)?.try_into()?;
+        let text: StrBuffer = kwargs.get(Qstr::MP_QSTR_text)?.try_into()?;
+
+        let text: String<50> = String::from(text.trim());
```

Comment:
this is a strong argument for separating type `T` of major/minor prompt from type of choices.

here you could push `title` and `text` as the major/minor prompt, since `StrBuffer` is `AsRef<str>`, and completely separately, choices as `[&str]`

Reply:
Done in [880cb15](https://github.com/trezor/trezor-firmware/pull/2277/commits/880cb1576cf15679cb98eee4181d1367a4fe929d) - the `title` and `subtitle` were moved to `Frame`, the choice components themselves to not show any description text